packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -0,0 +1,222 @@
+
+// TODO(Piinks): This file will replace ./client.dart when transition to Skia
+// Gold testing is complete
+
+const String _kGoldctlKey = 'GOLDCTL';
+const String _kServiceAccountKey = 'GOLD_SERVICE_ACCOUNT';
+
+/// An extension of the [GoldensClient] class that interfaces with Skia Gold
+/// for golden file testing.
+class SkiaGoldClient extends GoldensClient {
+  SkiaGoldClient({
+    FileSystem fs = const LocalFileSystem(),
+    ProcessManager process = const LocalProcessManager(),
+    Platform platform = const LocalPlatform(),
+  }) : super(
+    fs: fs,
+    process: process,
+    platform: platform,
+  );
+
+  /// The local [Directory] within the [comparisonRoot] for the current test
+  /// context. In this directory, the client will create image and json files
+  /// for the goldctl tool to use.
+  ///
+  /// This is informed by the [FlutterGoldenFileComparator] [basedir]. It cannot
+  /// be null.
+  Directory _workDirectory;
+
+  /// The path to the local [Directory] where the goldctl tool is hosted.
+  ///
+  /// Uses the [platform] environment in this implementation.
+  String get _goldctl => platform.environment[_kGoldctlKey];
+
+  /// The path to the local [Directory] where the service account key is
+  /// hosted.
+  ///
+  /// Uses the [platform] environment in this implementation.
+  String get _serviceAccount => platform.environment[_kServiceAccountKey];
+
+  @override
+  Directory get comparisonRoot => flutterRoot.childDirectory(fs.path.join('bin', 'cache', 'pkg', 'skia_goldens'));
+
+  /// Prepares the local work space for golden file testing and calls the
+  /// goldctl `auth` command.
+  ///
+  /// This ensures that the goldctl tool is authorized and ready for testing. It
+  /// will only be called once for each instance of
+  /// [FlutterSkiaGoldFileComparator].
+  ///
+  /// The [workDirectory] parameter specifies the current directory that golden
+  /// tests are executing in, relative to the library of the given test. It is
+  /// informed by the basedir of the [FlutterSkiaGoldFileComparator].
+  Future<void> auth(Directory workDirectory) async {
+    assert(workDirectory != null);
+    _workDirectory = workDirectory;
+    if (_clientIsAuthorized())
+      return;
+
+    if (_serviceAccount.isEmpty) {
+      final StringBuffer buf = StringBuffer()..writeln('Gold service account is unavailable.');
+      throw NonZeroExitCode(1, buf.toString());
+    }
+
+    final File authorization = _workDirectory.childFile('serviceAccount.json');
+    await authorization.writeAsString(_serviceAccount);
+
+    final List<String> authArguments = <String>[
+      'auth',
+      '--service-account', authorization.path,
+      '--work-dir', _workDirectory.childDirectory('temp').path,
+    ];
+
+    // final io.ProcessResult authResults =
+    await io.Process.run(
+      _goldctl,
+      authArguments,
+    );
+    // TODO(Piinks): Re-enable after Gold flakes are resolved, https://github.com/flutter/flutter/pull/36103
```

Comment:
Flaky behavior from the skia gold service needs to be resolved before these exceptions can be enabled.